packages/components/view-switcher/src/view-switcher-button.tsx

```diff
@@ -7,52 +7,66 @@ import {
 import { css } from '@emotion/react';
 import AccessibleButton from '@commercetools-uikit/accessible-button';
 import { designTokens } from '@commercetools-uikit/design-system';
+import { warning } from '@commercetools-uikit/utils';
 import { getButtonStyles } from './view-switcher.styles';
 
 export type TViewSwitcherButtonProps = {
-  isDisabled?: boolean;
+  children?: string;
+  icon?: ReactElement;
   isActive?: boolean;
-  children: string;
+  isDisabled?: boolean;
   isCondensed?: boolean;
   isFirstButton?: boolean;
   isLastButton?: boolean;
-  icon?: ReactElement;
+  label?: string;
   value: string;
   onClick?: (
     event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
   ) => void;
 };
 
-const ViewSwitcherButton = (props: TViewSwitcherButtonProps) => (
-  <AccessibleButton
-    label={props.children}
-    css={getButtonStyles(
-      props.isDisabled,
-      props.isActive,
-      props.isCondensed,
-      props.isFirstButton,
-      props.isLastButton
-    )}
-    onClick={props.onClick}
-    isDisabled={props.isDisabled}
-  >
-    {props.icon && (
-      <span
-        css={css`
-          margin: ${designTokens.marginForViewSwitcherIcon};
-          display: flex;
-          align-items: center;
-          justify-content: center;
-        `}
-      >
-        {cloneElement(props.icon, {
-          size: props.isCondensed ? 'medium' : 'big',
-        })}
-      </span>
-    )}
-    <span>{props.children}</span>
-  </AccessibleButton>
-);
+const ViewSwitcherButton = (props: TViewSwitcherButtonProps) => {
+  warning(
+    Boolean(props.children || props.icon),
+    'uikit/ViewSwitcherButton: You need to provide at least the `children` to render inside the button or an `icon`.'
+  );
+
+  warning(
+    Boolean(props.label || props.children),
```

Comment:
Ok, I looked at the conditions for 3m but it's late-ish: aren't both conditions the same?

Reply:
Previously the component had the `children` prop as required as we expected to always us it with some text. The `icon` property was optional.
Now we can use just an `icon` so `children` both props are optional, but we expect consumers to use at least one of them.
This is what the first condition is for.

Then, if consumer only wants to use an icon, we need him/her to provide a label for the icon (when using children, we use it as the label).
The second condition makes sure we have something for the button label.

Does it makes sense?

Reply:
Thanks. Got it.